packages/yarnpkg-pnpify/tests/RawHoister.test.ts

```diff
@@ -0,0 +1,81 @@
+import {RawHoister, PackageTree} from '../sources/RawHoister';
+
+const toTree = (obj: Record<string, number[]>) => new Map(
+  Object.entries(obj).map(
+    tuple => [Number(tuple[0]), new Set(tuple[1])]
+  )
+);
+
+const toObject = (tree: PackageTree): Record<string, number[]> =>
+  [...tree.entries()].reduce((obj: Record<string, number[]>,
+    [key, value]) => (obj[key] = [...value], obj), {});
+
+describe('RawHoister', () => {
+  const hoister = new RawHoister();
+
+  it('should do very basic hoisting', () => {
+    const tree = toTree({
```

Comment:
@arcanis I'm not sure I understand how new Map, will improve readability here, could you clarify?

Reply:
When I read this code I initially thought it was transforming the input into an actual nested tree. Then I had to look at the helper to understand it really was just doing:

```js
const tree = new Map([
  [0, new Set([1])],
  [1, new Set([2])],
]);
```

Since that's almost the same syntax as the one used to initialize `packageMap` a few lines below, I'm not sure the `toTree` helper adds a lot (compared to the need to check its implementation to understand what it does).

Reply:
@arcanis Agreed, removed `toTree` helper